### DISCUSSION TITLE
Problems configuring replication in minio

### DISCUSSION THREAD

**Discussion by jmbarreiro1** [minio/minio#19942] [Q&A] [answered]
Hello,

I have a problem with replication between two tenants deployed in two different kubernetes clusters.  The scenario is simple, tenant on source cluster, tenant on target cluster. To expose the tenant in the target cluster I am using a load balancer type service, adding a DNS entry that works perfectly when I do a mc alias set. On the source cluster I have done the same to test. The problem is that when I try to execute the command mc replicate add, using those urls that attack the tenant whatever it is, as long as it is through a load balancer type service I get a 503.

**Bucket replication**

mc replicate add source/main --remote-bucket target/main --priority 1 --insecure

mc: <ERROR> unable to configure remote target. Remote service connection error (Remote service endpoint offline, target bucket: main or remote service credentials: mfs-client-back-root invalid
	Get “https://mfs-client-back.target.com/main/?location=”: Forbidden).
In debug mode mc replicate add source/main --remote-bucket target/main  --priority 1 --insecure --debug
mc: <DEBUG> PUT /minio/admin/v3/set-remote-target?bucket=main HTTP/1.1
Host: [mfs-client-back.source.com](http://mfs-client-back.source.com/)
User-Agent: MinIO (linux; amd64) madmin-go/2.0.0 mc/RELEASE.2024-06-01T15-03-35Z
Content-Length: 587
Accept-Encoding: zstd,gzip
Authorization: AWS4-HMAC-SHA256 Credential=mfs-client-back-root/20240615//s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: 7ba39edd0295ac886c38d9dc426d1a90007680ab1d8f21372b94e8c0ab432921
X-Amz-Date: 20240615T111504Z
mc: <DEBUG> HTTP/1.1 503 Service Unavailable
Content-Length: 483
Accept-Ranges: bytes
Content-Type: application/json
Date: Sat, 15 Jun 2024 11:15:04 GMT
Server: MinIO
Strict-Transport-Security: max-age=31536000; includeSubDomains
Vary: Origin
Vary: Accept-Encoding
X-Amz-Id-2: 3ece9885cbaec066b8db661fe1120eba7c8fd03eb43c1a965fdddd741eab41d1
X-Amz-Request-Id: 17D92943EAE77761
X-Content-Type-Options: nosniff
X-Xss-Protection: 1; mode=block
{"Code":"XMinioAdminReplicationRemoteConnectionError","Message":"Remote service connection error (Remote service endpoint offline, target bucket: main or remote service credentials: mfs-client-back-root invalid \n\tGet \"https://mfs-client-back.target.com/main/?location=\": Forbidden)","BucketName":"main","Resource":"/minio/admin/v3/set-remote-target","RequestId":"17D92943EAE77761","HostId":"3ece9885cbaec066b8db661fe1120eba7c8fd03eb43c1a965fdddd741eab41d1"}
mc: <DEBUG> TLS Certificate found:
mc: <DEBUG>  >> Expires: 2024-07-23 13:05:46 +0000 UTC
mc: <DEBUG> TLS Certificate found:
mc: <DEBUG>  >> Expires: 2029-05-21 11:36:54 +0000 UTC
mc: <DEBUG> Response Time: 451.40375ms
mc: <DEBUG> PUT /minio/admin/v3/set-remote-target?bucket=main HTTP/1.1
Host: [mfs-client-back.apps.source.com](http://mfs-client-back.apps.source.com/)
User-Agent: MinIO (linux; amd64) madmin-go/2.0.0 mc/RELEASE.2024-06-01T15-03-35Z
Content-Length: 587
Accept-Encoding: zstd,gzip
Authorization: AWS4-HMAC-SHA256 Credential=mfs-client-back-root/20240615//s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: 7ba39edd0295ac886c38d9dc426d1a90007680ab1d8f21372b94e8c0ab432921
X-Amz-Date: 20240615T111504Z
mc: <DEBUG> HTTP/1.1 503 Service Unavailable
Content-Length: 483
Accept-Ranges: bytes
Content-Type: application/json
Date: Sat, 15 Jun 2024 11:15:04 GMT
Server: MinIO
Strict-Transport-Security: max-age=31536000; includeSubDomains
Vary: Origin
Vary: Accept-Encoding
X-Amz-Id-2: 3ece9885cbaec066b8db661fe1120eba7c8fd03eb43c1a965fdddd741eab41d1
X-Amz-Request-Id: 17D92943F736EDBE
X-Content-Type-Options: nosniff
X-Xss-Protection: 1; mode=block
{"Code":"XMinioAdminReplicationRemoteConnectionError","Message":"Remote service connection error (Remote service endpoint offline, target bucket: main or remote service credentials: mfs-client-back-root invalid \n\tGet \"https://mfs-client-back.target.com/main/?location=\": Forbidden)","BucketName":"main","Resource":"/minio/admin/v3/set-remote-target","RequestId":"17D92943F736EDBE","HostId":"3ece9885cbaec066b8db661fe1120eba7c8fd03eb43c1a965fdddd741eab41d1"}
mc: <DEBUG> TLS Certificate found:
mc: <DEBUG>  >> Expires: 2024-07-23 13:05:46 +0000 UTC
mc: <DEBUG> TLS Certificate found:
mc: <DEBUG>  >> Expires: 2029-05-21 11:36:54 +0000 UTC
mc: <DEBUG> Response Time: 162.901118ms
mc: <DEBUG> PUT /minio/admin/v3/set-remote-target?bucket=main HTTP/1.1
Host: [mfs-client-back.source.com](http://mfs-client-back.source.com/)
User-Agent: MinIO (linux; amd64) madmin-go/2.0.0 mc/RELEASE.2024-06-01T15-03-35Z
Content-Length: 587
Accept-Encoding: zstd,gzip
Authorization: AWS4-HMAC-SHA256 Credential=mfs-client-back-root/20240615//s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: 7ba39edd0295ac886c38d9dc426d1a90007680ab1d8f21372b94e8c0ab432921
X-Amz-Date: 20240615T111505Z
mc: <DEBUG> HTTP/1.1 503 Service Unavailable
Content-Length: 483
Accept-Ranges: bytes
Content-Type: application/json
Date: Sat, 15 Jun 2024 11:15:05 GMT
Server: MinIO
Strict-Transport-Security: max-age=31536000; includeSubDomains
Vary: Origin
Vary: Accept-Encoding
X-Amz-Id-2: 3ece9885cbaec066b8db661fe1120eba7c8fd03eb43c1a965fdddd741eab41d1
X-Amz-Request-Id: 17D92944277EF4E3
X-Content-Type-Options: nosniff
X-Xss-Protection: 1; mode=block
{"Code":"XMinioAdminReplicationRemoteConnectionError","Message":"Remote service connection error (Remote service endpoint offline, target bucket: main or remote service credentials: mfs-client-back-root invalid \n\tGet \"https://mfs-client-back.target.com/main/?location=\": Forbidden)","BucketName":"main","Resource":"/minio/admin/v3/set-remote-target","RequestId":"17D92944277EF4E3","HostId":"3ece9885cbaec066b8db661fe1120eba7c8fd03eb43c1a965fdddd741eab41d1"}
mc: <DEBUG> TLS Certificate found:
mc: <DEBUG>  >> Expires: 2024-07-23 13:05:46 +0000 UTC
mc: <DEBUG> TLS Certificate found:
mc: <DEBUG>  >> Expires: 2029-05-21 11:36:54 +0000 UTC
mc: <DEBUG> Response Time: 78.513105ms
mc: <DEBUG> PUT /minio/admin/v3/set-remote-target?bucket=main HTTP/1.1
Host: [mfs-client-back.source.com](http://mfs-client-back.source.com/)
User-Agent: MinIO (linux; amd64) madmin-go/2.0.0 mc/RELEASE.2024-06-01T15-03-35Z
Content-Length: 587
Accept-Encoding: zstd,gzip
Authorization: AWS4-HMAC-SHA256 Credential=mfs-client-back-root/20240615//s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: 7ba39edd0295ac886c38d9dc426d1a90007680ab1d8f21372b94e8c0ab432921
X-Amz-Date: 20240615T111507Z
mc: <DEBUG> HTTP/1.1 503 Service Unavailable
Content-Length: 483
Accept-Ranges: bytes
Content-Type: application/json
Date: Sat, 15 Jun 2024 11:15:07 GMT
Server: MinIO
Strict-Transport-Security: max-age=31536000; includeSubDomains
Vary: Origin
Vary: Accept-Encoding
X-Amz-Id-2: 3ece9885cbaec066b8db661fe1120eba7c8fd03eb43c1a965fdddd741eab41d1
X-Amz-Request-Id: 17D929448A16138C
X-Content-Type-Options: nosniff
X-Xss-Protection: 1; mode=block

**Site replication**

mc admin replicate add source3 target --insecure --debug

c: <ERROR> Unable to add sites for replication. Error received when contacting a peer site (unable to create admin client for target: Remote service endpoint https://mfs-client-back.target.com not available
	remote target is offline for endpoint https://mfs-client-back.target.com)
 (1) admin-replicate-add.go:122 cmd.mainAdminReplicateAdd(..) Tags: [source3, target]
 (0) admin-replicate-add.go:122 cmd.mainAdminReplicateAdd(..)
 Release-Tag:RELEASE.2024-06-01T15-03-35Z | Commit:3cfa8642fdab | Host:jmbarreiro | OS:linux | Arch:amd64 | Lang:go1.22.3 | Mem:68 MiB/83 MiB | Heap:68 MiB/75 MiB.

In the source I can see these logs

API: SiteReplicationAdd()
 Time: 06:26:24 UTC 06/17/2024
 DeploymentID: 7be60660-e3d1-44f0-8a9b-21665d7775d4
 RequestID: 17D9B6AC6FD05743
 RemoteHost: [::1]
 UserAgent: MinIO (linux; amd64) madmin-go/2.0.0 mc/RELEASE.2024-06-01T15-03-35Z
 Error: unable to create admin client for target: Remote service endpoint https://mfs-client-back.target.com not available
	remote target is offline for endpoint https://mfs-client-back.target.com (cmd.SRError)
        4: internal/logger/logger.go:268:logger.LogIf()
        3: cmd/logging.go:42:cmd.adminLogIf()
        2: cmd/admin-handlers-site-replication.go:57:cmd.adminAPIHandlers.SiteReplicationAdd()
        1: net/http/server.go:2166:http.HandlerFunc.ServeHTTP()

if I deploy two tenants in the same kubernetes cluster it works perfectly. Any idea?

Thanks a lot!

Regards,
Jose

**Comment by jmbarreiro1** [comment]
Hi,

any ideas?

Thanks a lot!

**Reply by klauspost** [reply, marked as answer]
> Remote service endpoint https://mfs-client-back.target.com/ not available
remote target is offline for endpoint https://mfs-client-back.target.com/

The target isn't reachable from there. So you need to fix that.
